mne/channels/tests/test_interpolation.py

```diff
@@ -76,8 +75,26 @@ def test_interplation():
     evoked.interpolate_bads_eeg()
     assert_array_equal(ave_after, evoked.data[bads_idx])
 
+    # make sure error is raised when preload is False
     for inst in [raw, epochs]:
         assert hasattr(inst, 'preload')
         inst.preload = False
         inst.info['bads'] = [inst.ch_names[1]]
         assert_raises(ValueError, inst.interpolate_bads_eeg)
+
+    # test interpolating different channels per epoch
+    epochsi = epochs_orig.copy()
+    epochs_12 = epochs_orig.copy()
+    epochs_12.info['bads'] = ['EEG 012']
+    epochs_12.interpolate_bads_eeg()
+    epochs_12_17 = epochs_orig.copy()
+    epochs_12_17.info['bads'] = ['EEG 012', 'EEG 017']
+    epochs_12_17.interpolate_bads_eeg()
+    _interpolate_bads_eeg_epochs(epochsi, [['EEG 012'], [],
+                                           ['EEG 012', 'EEG 017'], ['EEG 012'],
+                                           [], [], []])
+    assert_allclose(epochsi._data[0], epochs_12._data[0])
+    assert_array_equal(epochsi._data[1], epochs_orig._data[1])
+    assert_allclose(epochsi._data[2], epochs_12_17._data[2])
+    assert_allclose(epochsi._data[3], epochs_12._data[3])
```

Comment:
Yes that sounds reasonable, when I recast the single epoch data I get exact matches (https://github.com/christianbrodbeck/mne-python/commit/557d9da7df9c2b7ceb6df502593ab4befda293cf)